services/replay/service.go

```diff
@@ -780,9 +799,22 @@ func (s *Service) handleReplay(w http.ResponseWriter, req *http.Request) {
 	}
 	replay, err := s.replays.Get(id)
 	if err != nil {
-		httpd.HttpError(w, "could not find replay: "+err.Error(), true, http.StatusNotFound)
+		httpd.HttpError(w, "could not find replay: "+id, true, http.StatusNotFound)
+		return
+	}
+
+	finished := req.URL.Query().Get("finished")
+	if finished == "true" && replay.Status != Finished {
+		httpd.HttpError(w, "replay status is not Finished", true, http.StatusBadRequest)
+		return
+	}
+
+	if finished == "true" && replay.Status == Finished {
```

Comment:
Seems like we don't need this branch, see above comment about always adding the stats to the response.

Reply:
Yep, will remove.